externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteScala.scala

```diff
@@ -64,6 +68,27 @@ class ExecuteScala(
       if (legacyOutput.nonEmpty) {
         warn(s"Clearing legacy output from last interpreting:\n $legacyOutput")
       }
+
+      val jars = spark.sparkContext.listJars().flatMap { jar =>
+        val jarFile = new File(jar)
+        spark.sharedState.jarClassLoader.getURLs.groupBy(_.getFile.split("/").last).get(
+          jarFile.getName) match {
```

Comment:
Should we use URL?
```
        val jarName = new URL(jar).getFile.split("/").last
        spark.sharedState.jarClassLoader.getURLs.groupBy(_.getFile.split("/").last).get(
          jarName) match
```

Reply:
`SparkContext.listJars` are of type `spark://` scheme, URL that does not support this type

Reply:
got it